xbmc/games/addons/GameClient.cpp

```diff
@@ -713,6 +717,58 @@ bool CGameClient::Deserialize(const uint8_t* data, size_t size)
   return bSuccess;
 }
 
+void CGameClient::LoadSRAM()
+{
+  std::string path = URIUtils::AddFileToFolder(CProfilesManager::GetInstance().GetSavestatesFolder(), "SRAM");
+              path = URIUtils::AddFileToFolder(path, URIUtils::GetFileName(m_gamePath));
+
+  uint8_t *data;
+  size_t size;
+  m_pStruct->GetMemory(GAME_MEMORY_SAVE_RAM, &data, &size);
+
+  std::string sramPath = path + ".sram";
+  XFILE::CFile sramFile;
+  if (sramFile.Open(sramPath))
+  {
+    ssize_t read = sramFile.Read(data, size);
+    if (read != (ssize_t)size)
+    {
+      CLog::Log(LOGERROR, "GAME: Failed read from SRAM file %s", sramPath.c_str());
```

Comment:
Adding the size in bytes might be helpful, we might as well take advantage of knowing it